### PR TITLE
Add support for CN0556

### PR DESCRIPTION
**Pull request by AndyOesmer**
# Description

++ Adding support for CN0556 
- Added cn0556.py in library
- Added cn0556 examples (Buck Mode Example and Boost Mode Example)
- Added cn0556 production test scripts
- Added pytest for cn0556

- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

The example scripts were run by using interfacing the EVAL-CN0556-EBZ with an EVAL-CN0554-RPIZ and a Raspberry Pi 4 connection. The SD card of the Raspberry Pi 4 is flashed with the ADI Kuiper Linux image. The device tree overlay must be configured to identify the CN0556 connected to it.

**Test Configuration**:
* Hardware: EVAL-CN0556-EBZ + EVAL-CN0554-RPIZ + Raspberry Pi 4 with ADI Kuiper Linux Image
* OS: pyadi-iio examples can be run directly from Raspberry Pi GUI or by accessing remotely thru LAN

# Documentation

Product Page: https://www.analog.com/en/design-center/reference-designs/circuits-from-the-lab/cn0556.html
User Guide: https://wiki.analog.com/resources/eval/user-guides/circuits-from-the-lab/cn0556

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by kister-jimenez** [comment]
Add this part to the supported_parts.md and rebase to the latest main.

File: examples/cn0556/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# CN0556
+## 1. Install Prerequisite:
+
+```cmd
+pip install -r requirements.txt
+pip install -e .
```
Comment (kister-jimenez) [inline examples/cn0556/README.md:6]: This is already part of the main README instructions.

File: examples/cn0556/requirements.txt
Comment (kister-jimenez) [inline examples/cn0556/requirements.txt:1]: These 2 are built-in python modules. You don't need this requirement.txt file.

File: adi/__init__.py [hunk 2]
```diff
@@ -103,6 +103,7 @@
 from adi.one_bit_adc_dac import one_bit_adc_dac
 from adi.QuadMxFE_multi import QuadMxFE_multi
 from adi.tdd import tdd
+from adi.cn0556 import cn0556
```
Comment (kister-jimenez) [inline adi/__init__.py:106]: Move this to the correct line or run pre-commit to sort the imports.

File: examples/cn0556/README.md (hunk 1 above)
Comment (AndyOesmer) [reply examples/cn0556/README.md]: Removed the step 1 - requirements installation step in README file.

File: examples/cn0556/requirements.txt
Comment (AndyOesmer) [reply examples/cn0556/requirements.txt:1]: Removed the requirements.txt file.

File: adi/__init__.py (hunk 2 above)
Comment (AndyOesmer) [reply adi/__init__.py]: Moved to correct line.

**Comment by tfcollins** [comment]
Please fix linting issues

**Comment by AndyOesmer** [comment]
> Please fix linting issues

Done.

**Review by mthoren-adi — APPROVED** [review]
@AndyOesmer , @kister-jimenez , looks like all of my structural suggestions to Andy were implemented, nice work.

**Pull request merged**